apps/src/code-studio/progressRedux.js

```diff
@@ -340,15 +341,19 @@ const userProgressFromServer = (state, dispatch, userId = null) => {
     }
 
     // We are on an overview page if currentLevelId is undefined.
-    const onOverviewPage = !state.currentLevelId;
+    const onOverviewPage = !progress.currentLevelId;
     // Show lesson plan links and other teacher info if teacher and on unit overview page.
     if (
       (data.isTeacher || data.teacherViewingStudent) &&
       !data.professionalLearningCourse &&
       onOverviewPage
     ) {
-      // Default to summary view if teacher is viewing their student, otherwise default to detail view.
-      dispatch(setIsSummaryView(data.teacherViewingStudent));
+      // Default to summary view if teacher is viewing their student or viewing student view, otherwise default to detail view.
+      dispatch(
+        setIsSummaryView(
+          data.teacherViewingStudent || viewAs === ViewType.Student
```

Comment:
I believe `teacherViewingStudent` is for when a teacher is viewing the progress of a specific student. `viewAs === ViewType.Student` means the teacher is looking at the lessons from the student view (but not looking at progress for a specific student)

Reply:
Ah, ok.  I tested this out and see what these two values control now.  Follow-up question: Why do we want to make this change?  (FWIW, when playing around this, I couldn't really figure out what the expected behavior was supposed to be. @fontie715)

Reply:
I originally made this change because when switch to student view it shows summary but when I load the page on student view it shows detail. So this would provide some consistency, but I think now that I better understand what summary view is, it's not necessary - I could remove it (maybe that would be safer)

Reply:
I agree with reverting this change but it'd probably be good to have a quick follow-up with Amanda about your observations around what is inconsistent and see if there are any changes we want to make there (on a separate ticket/PR).